`makesetup` to allow missing ending line break for `Setup` files

# Feature or enhancement

### Proposal:

When writing C/C++ extensions accoding to the [doc](https://docs.python.org/3/extending/extending.html), there would be problem if the `Setup.local` does not have an ending line break:
```log
bad word spammodule.o*doconfig* in spammodule.o*doconfig*
make: *** [Makefile] Error 1
```
The reason is that, the `*doconfig*` in the following location may be concatenated with the previous `$i`, thus forming an invalid line:
https://github.com/python/cpython/blob/2982bdb936f76518b29cf7de356eb5fafd22d112/Modules/makesetup#L95
So far the doc does not seem to explicitly mention adding line break. I think this can be fixed by checking for ending line break after the `cat`, and add one if there is not.



### Has this already been discussed elsewhere?

No response given

### Links to previous discussion of this feature:

_No response_

<!-- gh-linked-prs -->
### Linked PRs
* gh-116945
* gh-139100
<!-- /gh-linked-prs -->

--- a/Modules/makesetup
+++ b/Modules/makesetup
@@ -90,6 +90,7 @@ NL='\
 # Main loop
 for i in ${*-Setup}
 do
+	echo ''  # Add a linebreak so we don't choke on files missing EOL.
 	case $i in
 	-n)	echo '*noobjects*';;
 	*)	echo '*doconfig*'; cat "$i";;

